src/stan_math_backend/Statement_gen.ml

```diff
@@ -57,9 +57,12 @@ let rec integer_el_type = function
 
 let pp_decl ppf (vident, ut, adtype) =
   let pp_type =
-    if Transform_Mir.is_opencl_var vident then fun ppf _ ->
+    if
+      Transform_Mir.is_opencl_var vident
+      && ut <> UnsizedType.UInt && ut <> UnsizedType.UReal
+    then fun ppf _ ->
       match ut with
-      | UnsizedType.UInt | UArray UInt -> pf ppf "matrix_cl<int>"
+      | UArray UInt -> pf ppf "matrix_cl<int>"
```

Comment:
We do not handle that right now (need to add Math support to move arrays of matrices to arrays of matrix_cl.

Reply:
Will just add that to Math to prevent this behavior being weird right here.

Reply:
And I guess we expect the `is_opencl_var vident` to be false for any UArray UArray UInt, is that right? If so maybe we should add a case here for that to raise an exception instead of generating `matrix_cl<double>` for it? Don't feel too strongly.

Reply:
Have to double check. It should not pickup `UArray UArray UInt `yes and we should add an exception for that case. Once Math supports it I will open another PR that will also support that case. Does that work for you?

Reply:
Yep, that works!